src/js/media-error-dialog.ts

```diff
@@ -0,0 +1,108 @@
+import { globalThis } from './utils/server-safe-globals.js';
+import { MediaUIAttributes } from './constants.js';
+import { errors } from './labels/labels.js';
+import { MediaChromeDialog } from './media-chrome-dialog.js';
+import {
+  getNumericAttr,
+  getStringAttr,
+  setNumericAttr,
+  setStringAttr,
+} from './utils/element-utils.js';
+
+function getSlotTemplateHTML(attrs: Record<string, string>) {
+  return /*html*/ `
+    <style>
+      #content {
+        background: var(--media-control-background, var(--media-secondary-color, rgb(20 20 30 / .8)));
+        display: block;
+        padding: 1.2em 1.5em;
+      }
+
+      h3,
+      p,
+      ::slotted(h3),
+      ::slotted(p) {
+        margin-block: 0 .3em;
+      }
+    </style>
+    <slot name="error-${attrs.mediaerrorcode}" id="content">
+      ${getErrorMessage(attrs.mediaerrorcode, attrs.mediaerrormessage)}
+    </slot>
+  `;
+}
+
+function shouldHideErrorDialog(errorCode?: number) {
+  return !errorCode || errors[errorCode] === null;
+}
+
+function getErrorMessage(errorCode?: number | string, errorMessage?: string) {
+  const message: string = errors[+errorCode] ?? errorMessage ?? '';
+  const parts = message.split(':', 2);
+
+  if (parts.length === 2) {
+    return /*html*/ `
+      <h3>${parts[0]}</h3>
+      <p>${parts[1]}</p>
+    `;
+  }
+  return /*html*/ `<p>${message}</p> `;
+}
+
+const observedAttributes: string[] = [
+  MediaUIAttributes.MEDIA_ERROR_CODE,
+  MediaUIAttributes.MEDIA_ERROR_MESSAGE,
+];
+
+/**
+ * @attr {number} mediaerrorcode - (read-only) The error code for the current media error.
+ * @attr {string} mediaerrormessage - (read-only) The error message for the current media error.
+ *
+ * @cssproperty --media-control-background - `background` of control.
+ */
+class MediaErrorDialog extends MediaChromeDialog {
+  static getSlotTemplateHTML = getSlotTemplateHTML;
```

Comment:
question: Any reason we're implementing this as a class-level instead of an instance level that could be overridden?

Reply:
yes, it's on the class level so it could be called by React for example to render DSD (declarative shadow dom) without having to create an instance of this element.

overriding is still possible by child classes by redeclaring the static method.